R/aaa-auto.R

```diff
@@ -819,14 +819,14 @@ get_k_shortest_paths_impl <- function(graph, weights=NULL, k, from, to, mode=c("
   res <- .Call(R_igraph_get_k_shortest_paths, graph, weights, k, from-1, to-1, mode)
```

Comment:
I wonder why we have `-1` here but no inverse correction for the output. Is this expected?

Reply:
I don't have the code in front of me right now, but I remember that there aren't any clear guidelines on where to do the +1/-1 adjustments for vertex/edge indices and sometimes it's done in the output conversion of the result. For instance, there is `R_igraph_vector_int_to_SEXP` to convert an `igraph_vector_int_t` back to R without adjustment and `R_igraph_vector_int_to_SEXPp1` to convert it back _with_ adjustments. (The `p1` suffix stands for "plus one").

Ideally, there should be guidelines on where the adjustment is done. If it is easier to do in R and there are no performance implications (like an unnecessary copying of the vector), it should probably be done in R and we could get rid of the `p1` functions in `rinterface_extra.c`.

Reply:
There are performance implications, ideally we'd convert everything in C.

Reply:
Dreaming big: If we could get views that cast to int and perform the "minus one" on the fly, in the C core...